.github/workflows/release.yml

```diff
@@ -0,0 +1,46 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+*'
+
+env:
+  GOPROXY: https://proxy.golang.org/
+  PROTOCOL_VERSION: "5.0"
+  PROTOCOL_VERSIONS: "5.0"
+  SIGNER: interim_signing_subkey_7685B676
+
+jobs:
+  Release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Read go version
+        id: go-version
+        run: |
+          content=`cat ./.go-version`
+          echo "::set-output name=content::$content"
+      - terraform-provider-release:
+        name: 'Terraform Provider Release'
+        needs: go-version
+        uses: bflad/ghaction-terraform-provider-release/.github/workflows/hashicorp.yml@main
```

Comment:
Reusable workflows are a little weird because they must be declared at the job level: https://docs.github.com/en/actions/learn-github-actions/reusing-workflows#calling-a-reusable-workflow

So in this case, the `jobs` section of the workflow will look something like:

```yaml
jobs:
  go-version:
    runs-on: ubuntu-latest
    outputs:
      version: ${{ steps.go-version.outputs.version }}
    steps:
      - uses: actions/checkout@v2
      - id: go-version
        run: echo "::set-output name=version::$(cat ./.go-version)"
  terraform-provider-release:
    name: 'Terraform Provider Release'
    needs: go-version
    uses: bflad/ghaction-terraform-provider-release/.github/workflows/hashicorp.yml@main
    # ... further configuration of this job ...
```